CmsManager/CmsManagerSelector.php

```diff
@@ -79,7 +78,7 @@ public function isEditor()
      */
     public function onSecurityInteractiveLogin(InteractiveLoginEvent $event)
     {
-        if ($this->getSecurityContext()->getToken() &&
+        if ($this->container->get('security.token_storage')->getToken() &&
```

Comment:
It would be great to start using dependency injection instead of using this ugly container.

Reply:
good point! I will take a look at it!

Reply:
@Soullivaneuh is it considered a BC break if I add extra arguments to this service? Or is this allowed?

Reply:
Well, maybe adding the service with `null` value as default and trigger a deprecated error if it's not provided?

But maybe it's quite overkill as this class should be only use as a service.

cc @sonata-project/contributors for more opinions.

Reply:
Well, maybe add the argument with `null` value as default and trigger a deprecation error if not provided?

Maybe quite overkill. @sonata-project/contributors toughs?

Reply:
Maybe we should move the "service container cleanup" to a separate pr?

Reply:
I would remove the type hint on container, remove the argument, and throw if the provided argument is not a token storage. Also, I'd add a next major instruction asking for a type hint.

Reply:
If it's ok for you guys I will create a new issue to do this cleanup later on?

Reply:
It is for me

Reply:
Let's do that, smaller PR's are easier to review and merge

Reply:
> I would remove the type hint on container, remove the argument, and throw if the provided argument is not a token storage. Also, I'd add a next major instruction asking for a type hint.

As this class only needs the `security.token_storage` service, we can indeed do that. :+1:

Reply:
It does need more services no? There are a lot `$this->container->get('')` calls in this service

Reply:
I didn't see that, but indeed, you're right. Maybe add one argument per service, without type hinting, and make them optional, and then play with `func_get_args` /  `func_num_args` to simulate 2 different signatures?

Reply:
Yeah I'm looking at symfony how they do it now, adding/removing arguments. But indeed it will be something with `func_num_args`/`func_get_args`. I will come up with something and open a PR for it!